packages/vite-node/src/hmr/hmr.ts

```diff
@@ -160,8 +160,8 @@ async function fetchUpdate(
   }
 }
 
-function warnFailedFetch(err: Error, path: string | string[]) {
-  if (!err.message.match('fetch')) {
+function warnFailedFetch(err: any, path: string | string[]) {
```

Comment:
Using `unknown` would've caught the bug:

```suggestion
function warnFailedFetch(err: unknown, path: string | string[]) {
```